### DISCUSSION TITLE
How to get the coordinates for "Vacuum go to" service call

### DISCUSSION THREAD

**Discussion by mvdwetering** [humbertogontijo/homeassistant-roborock#187] [General]
I would like to use the "Vacuum go to" service call, but I can't figure out where to get the x and y coordinates I need to put in.
Anyone know where to find them?

**Comment by Ruppluka** [comment]
Do you mean vacuum_goto or vacuum_goto_predefined? 
To find the predefined outline point coordinates you can use Pin&Go and select the needed point on the map. You can see the coordinates on the visual editor in selection section.

**Comment by mvdwetering** [comment]
I only have a `roborock.vacuum_goto`. I don't see a vacuum_goto or vacuum_goto_predefined in the list.

![image](https://user-images.githubusercontent.com/732514/221410170-0c4fbcb8-534c-4bf4-be3d-c014d85125e7.png)

I am not sure what you mean with the Pin&Go on the map and where to find the visual editor. I tried to Pin&Go with the app, but could not find coordinates.

**Comment by Ruppluka** [comment]
You need to extract the Roborock map first by installing HACS integration "Xiaomi-vacuum-map-card"! Then you can create a custom card where you see the map and coordinates for your roborock.

**Comment by mvdwetering** [comment]
Sorry for the late response, only just had time to try. 

That really helped, specifically this page in the "Xiaomi-vacuum-map-card" documentation https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/blob/master/docs/templates/setup.md#getting-coordinates

I used the "Setup integer" approach with Pin-and-go and the 2 numbers I got could be used as X and Y coordinates.

**Comment by mvdwetering** [comment]
Actually it turns out there is another way without additional custom card.

When looking at the attributes of the Camera entity with the Developer Tools menu in Home Assistant there is a section near the bottom that has the current position of the robot. So you can drive the robot somewhere and then write down the numbers.

![image](https://user-images.githubusercontent.com/732514/228597658-e9929ed7-0854-4581-99af-eb9951a5386d.png)
